cmd/controller/main.go

```diff
@@ -27,20 +27,15 @@ import (
 )
 
 func main() {
-	options, manager := operator.NewOptionsWithManagerOrDie()
-	cloudProvider := cloudprovider.CloudProvider(awscloudprovider.NewCloudProvider(options.Ctx, cloudprovider.Options{
-		ClientSet:  options.Clientset,
-		KubeClient: options.KubeClient,
-		StartAsync: options.StartAsync,
-	}))
-	if hp, ok := cloudProvider.(operator.HealthCheck); ok {
-		utilruntime.Must(manager.AddHealthzCheck("cloud-provider", hp.LivenessProbe))
-	}
-	cloudProvider = cloudprovidermetrics.Decorate(cloudProvider)
-	if err := operator.RegisterControllers(options.Ctx,
-		manager,
-		controllers.GetControllers(options, cloudProvider)...,
-	).Start(options.Ctx); err != nil {
+	ctx := operator.NewContextOrDie()
+	cloudProvider := awscloudprovider.NewCloudProvider(ctx, cloudprovider.Options{
+		ClientSet:  ctx.Clientset,
+		KubeClient: ctx.KubeClient,
+		StartAsync: ctx.StartAsync,
+	})
+	utilruntime.Must(ctx.Manager.AddHealthzCheck("cloud-provider", cloudProvider.LivenessProbe))
```

Comment:
Can we remove the `HealthCheck` interface now?

Reply:
Used all over, unfortunately.